[firebase-analytics] Add pure JS analytics for use on Expo Client (#7093)
expo/expo: docs/pages/versions/unversioned/sdk/firebase-analytics.md

and linked into your app. Since the standard Expo Client loads projects on demand, it does not have
the google-services configuration linked into its app-bundle.

> Instead, the standard Expo Client relies on a pure JavaScript implementation of Firebase Analytics

Reviewer: ```suggestion
Instead, the standard Expo client relies on a JavaScript-based implementation of Firebase Analytics
```

to log events. This means that certain native lifecyle events are not recorded in the standard client,

> but you can still use `logEvent` to log events.

Reviewer: ```suggestion
but you can still use `logEvent` to record events.
```

> In order to use Firebase Analytics on the standard Client, ensure that the firebase web-configuration

Reviewer: ```suggestion
You may want to use Firebase Analytics in Expo client to verify that you are logging events at the time you intend to and with the data that you want to attach without having to do a standalone app build. To do set this up, ensure that the Firebase web configuration
```

is set in `app.json` and that `measurementId` exists in your firebase config. If `measurementId` 

> doesn't exist, then you need to enable or update Google Analytics in your Firebase project.

Reviewer: collapse paragraphs into one line so we don't add unnecessary linebreaks to docs

**app.json**
```json
{

>   "web": {

Reviewer: might want to nest this under the `"expo"` key to be extra clear
